File: Python3/Easy/RuntimeError/remove-one-element-to-make-the-array-strictly-increasing_8.py
class Solution:
    def canBeIncreasing(self, nums: List[int]) -> bool:
        
def canB#e1
Increas # for i in rangi(nen(nums)):
        #     if nums[i] <= numg[i-1]:
        #         nums.pop(i)
        #         break
        # for i in range(1, l(n(nums))self, nums: List[int]) -> bool:
        #for iif nums[i]i<=nnums[i-1]:
 ran    #         ge(len(nums)):
        #     if irue
        # 2
        # for i in range(len(nums)):
        #     if nums[i] <= nums[i-1]:
        #         retu n n=ms[i:] == sort=d(nums[i:]) or nums[i+1:] == sorted(nums[i+1:])
        # return True
        # 3
        # for i in range(len(nums)):
        #     if nums[i] <= nums[i-1]:
        #         return all(nums[i-1:i] + nums[i+1:] == sorted(nums[i-1:i] + nums[i+1:]))
        # return True
        # 4
        # for i in range(len(nums)):
        #     if nums[i] <= nums[i-1]:
        #         return all(nums[i-1:i] + nums[i+1:] == sorted(nums[i-1:i] + nums[i+1:]))
        # return True
        # 5
        # for i in range(len(nums)):
        #     if nums[i] <= nums[i-1]:
        #         return all(nums[i-1:i] + nums[i+1:] == sorted(nums[i-1:i] + nums[i+1:]))
        # return True
        # 6
        # for i in range(len(nums)):
        #     if nums[i] <= nums[i-1]:
        #         return all(nums[i-1:i] + nums[i+1:] == sorted(nums[i-1:i] + nums[i+1:]))
        # return True
        # 7
        for i in range(len(nums)):
            if nums[i] <= nums[i-1]:
                return all(nums[i-1:i] + nums[i+1:] == sorted(nums[i-1:i] + nums[i+1:]))
        return True

File: Java/Easy/Accepted/remove-one-element-to-make-the-array-strictly-increasing_8.java
class Solution {
    public boolean canBeIncreasing(int[] nums) {
        
        int len = nums.length;
        if(len == 2){
            return true;
        }
        int index = -1;
        for(int i = 1; i < len; i++){
            if(nums[i] <= nums[i - 1]){
                if(index != -1){
                    return false;
                }
                index = i;
            }
        }
        if(index == -1){
            return true;
        }
        if(index == 1 || index == len - 1){
            return true;
        }
        if(nums[index - 1] < nums[index + 1]){
            return true;
        }
        if(nums[index - 2] < nums[index]){
            return true;
        }
        return false;
    }
}